dev/requirements.txt

```diff
@@ -1,5 +1,10 @@
-flake8==3.5.0
+flake8==3.7.*
```

Comment:
While I still think we should pin every version here, perhaps this approach is a compromise we can agree on.

`==3.7.*` means pip will install the latest bugfix release on 3.7. If you already have any 3.7 version installed, even if it's not the latest one, pip will consider the requirement satisfied and won't do anything. To force pip to upgrade to the latest bugfix release of 3.7 when you already have a compatible version installed, simply request it via `pip install --upgrade`.

Reply:
If we can assume these dependencies follow SemVer, it we should better use wildcards on minor versions ...

Reply:
I mentioned it elsewhere but I'll mention it again here: Linters like flake8 and pycodestyle introduce new checks in minor/feature releases. There is very high chance that every new check they introduce will flag new problems and fail the build. (In fact, we saw exactly that behavior with pydocstyle just before we removed it.)

I don't understand the point of waiting for the build to break before pinning or severely limiting the versions for libraries like these.